core/store/src/trie/iterator.rs

```diff
@@ -422,12 +474,19 @@ mod tests {
         assert_eq!(result1, result2);
     }
 
-    fn test_seek(trie: &Trie, map: &BTreeMap<Vec<u8>, Vec<u8>>, seek_key: &[u8]) {
+    fn test_seek_prefix(trie: &Trie, map: &BTreeMap<Vec<u8>, Vec<u8>>, seek_key: &[u8]) {
         let mut iterator = trie.iter().unwrap();
-        iterator.seek(&seek_key).unwrap();
-        let result1: Vec<_> = iterator.map(Result::unwrap).take(5).collect();
-        let result2: Vec<_> =
-            map.range(seek_key.to_vec()..).map(|(k, v)| (k.clone(), v.clone())).take(5).collect();
+        iterator.seek_prefix(&seek_key).unwrap();
+        let iterator = iterator.map(Result::unwrap).inspect(|(key, _)| {
+            assert!(key.starts_with(seek_key), "‘{key:x?}’ does not start with ‘{seek_key:x?}’");
+        });
+        let result1: Vec<_> = iterator.take(5).collect();
```

Comment:
I’d still like to test all the entries for whether the prefix is correct.  How about:

```suggestion
        let got = Vec::with_capacity(5);
        for item in iterator {
            let (key, value) = item.unwrap();
            assert!(key.starts_with(seek_key), "‘{key:x?}’ does not start with ‘{seek_key:x?}’");
            if got.len() < 5 {
                got.push((key, value));
            }
        }
```

And then could you also rename `result2` to `want` below and the final
assert would then be `assert_eq!(want, got);`.